geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteRegionFunctionOp.java

```diff
@@ -63,48 +63,29 @@ private ExecuteRegionFunctionOp() {
     // no instances allowed
   }
 
-  /**
-   * Does a execute Function on a server using connections from the given pool to communicate with
-   * the server.
-   *
-   * @param pool the pool to use to communicate with the server.
-   * @param region the name of the region to do the put on
-   * @param function to be executed
-   * @param serverRegionExecutor which will return argument and filter
-   * @param resultCollector is used to collect the results from the Server
-   * @param timeoutMs timeout in milliseconds
-   */
-  public static void execute(ExecutablePool pool, String region, Function function,
+  private static void execute(ExecutablePool pool, String region, String function,
       ServerRegionFunctionExecutor serverRegionExecutor, ResultCollector resultCollector,
-      byte hasResult, int mRetryAttempts, final int timeoutMs) {
-
-    ExecuteRegionFunctionOpImpl op =
-        new ExecuteRegionFunctionOpImpl(region, function, serverRegionExecutor,
-            resultCollector, hasResult, new HashSet<>(), timeoutMs);
-    boolean reexecute = false;
-    boolean reexecuteForServ = false;
-    Set<String> failedNodes = new HashSet<>();
-    AbstractOp reexecOp;
-
-    int maxRetryAttempts = mRetryAttempts;
-    if (!function.isHA()) {
+      byte hasResult, int maxRetryAttempts, boolean isHA, boolean optimizeForWrite,
+      AbstractOp op, boolean isReexecute, Set<String> failedNodes) {
+
+    if (!isHA) {
       maxRetryAttempts = 0;
     }
 
     do {
       try {
-        if (reexecuteForServ) {
-          reexecOp = new ExecuteRegionFunctionOpImpl(op,
+        if (isReexecute) {
+          op = new ExecuteRegionFunctionOpImpl((ExecuteRegionFunctionOpImpl) op,
```

Comment:
If you change the type of `op` to `ExecuteRegionFunctionOpImpl` then this blind cast can go away. Since it’s the only type that can be passed into this method without causing a cast exception it should just be the parameter type.

Reply:
Thanks